ivy.el

```diff
@@ -2955,13 +2953,13 @@ All CANDIDATES are assumed to match NAME."
   "Re-sort candidates by NAME.
 All CANDIDATES are assumed to match NAME.
 Prefix matches to NAME are put ahead of the list."
-  (if (or (string-match "^\\^" name) (string= name ""))
+  (if (or (string-match-p "^\\^" name) (string= name ""))
```

Comment:
Should the regexp here be:
```el
"\\`\\^"
```
?

Reply:
Well, if there's only a single line in the `name` string then there would be no difference. I don't have a strong opinion in this case so I left it as is since it appears to be working OK.

Reply:
> Well, if there's only a single line in the `name` string then there would be no difference.

Yes, but if this code shouldn't be matching line beginnings then it shouldn't be matching line beginnings.